File: image/rotate_cv.py
import cv2

# 이미지 열기
image_path = 'image1.jpg'  # 이미지 파일 경로를 입력하세요
image = cv2.imread(image_path)

# 이미지가 제대로 로드되었는지 확인
if image is None:
    raise ValueError(f"Failed to load the image from {image_path}.")
    
# 이미지 회전 (시계 반대 방향으로 90도)
rotated_image = cv2.rotate(image, cv2.ROTATE_90_COUNTERCLOCKWISE)

# 회전된 이미지 저장
rotated_image_path = 'rotated_cv_image.jpg'  # 저장할 파일 경로를 입력하세요
cv2.imwrite(rotated_image_path, rotated_image)

# 회전된 이미지 표시 (옵션)
cv2.imshow('Rotated Image', rotated_image)
cv2.waitKey(0)
cv2.destroyAllWindows()

File: image/rotate_cv.c
#include <opencv4/opencv.h>
#include <stdio.h>

int main(int argc, char** argv) {
    // 이미지 파일 경로
    const char* imagePath = "image.jpg";  // 이미지 파일 경로를 입력하세요
    const char* outputPath = "rotated_image.jpg";  // 저장할 파일 경로를 입력하세요

    // 이미지 로드
    IplImage* image = cvLoadImage(imagePath, CV_LOAD_IMAGE_COLOR);
    if (!image) {
        fprintf(stderr, "Error: Unable to load the image.\n");
        return -1;
    }

    // 이미지 회전 (시계 반대 방향으로 90도)
    IplImage* rotatedImage = cvCreateImage(cvSize(image->height, image->width), IPL_DEPTH_8U, image->nChannels);
    cvTranspose(image, rotatedImage);
    cvFlip(rotatedImage, rotatedImage, 1);  // 1은 수평으로 flip

    // 회전된 이미지 저장
    if (!cvSaveImage(outputPath, rotatedImage, 0)) {
        fprintf(stderr, "Error: Unable to save the rotated image.\n");
        cvReleaseImage(&image);
        cvReleaseImage(&rotatedImage);
        return -1;
    }

    // 이미지 해제
    cvReleaseImage(&image);
    cvReleaseImage(&rotatedImage);

    printf("Image rotated and saved successfully.\n");
    return 0;
}
